### PR TITLE
Some cmake build improvements

### PR DESCRIPTION
**Pull request by kilograham**
- Change some cmake output to DEBUG level
- Make SDK build more consistent with other libraries (use an INTERFACE marker library for inclusion tests)
- Add PICO_SDK_PRE_LIST_FILES, PICO_SDK_POST_LIST_FILES build vars

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -24,6 +28,7 @@ if (NOT TARGET _pico_sdk_inclusion_marker)
 
     # allow customization
     add_sub_list_dirs(PICO_SDK_PRE_LIST_DIRS)
+    add_sub_list_dirs(PICO_SDK_PRE_LIST_FILES)
```
Comment (lurch) [inline CMakeLists.txt:31]: Was this supposed to be `add_sub_list_files` as you've done on line 40? :shrug:

File: pico_sdk_init.cmake [hunk 2]
```diff
@@ -1,54 +1,64 @@
...
-include(pico_pre_load_toolchain)
+    include(pico_pre_load_platform)
 
-macro(pico_sdk_init)
-    if (NOT CMAKE_PROJECT_NAME)
-        message(WARNING "pico_sdk_init() should be called after the project is created (and languages added)")
-    endif()
-    add_subdirectory(${PICO_SDK_PATH} pico-sdk)
-endmacro()
+    # todo perhaps this should be included by the platform instead?
```
Comment (lurch) [inline pico_sdk_init.cmake:38]: It's a shame that the GitHub web-diff-viewer gets so badly confused by a change-of-indent :confused:

File: CMakeLists.txt (hunk 1 above)
Comment (kilograham) [reply CMakeLists.txt:31]: yes; oops

File: pico_sdk_init.cmake (hunk 2 above)
Comment (kilograham) [reply pico_sdk_init.cmake:38]: there is a "hide whitespace changes" under the gear icon (which you can flip temporarily)

Comment (lurch) [reply pico_sdk_init.cmake:38]: Ahh, that's super handy, thanks! :heart:

**Review by liamfraser — APPROVED** [review]

File: pico_sdk_init.cmake [hunk 3]
```diff
@@ -1,54 +1,64 @@
...
 
-macro(add_sub_list_dirs var)
-    foreach(LIST_DIR IN LISTS ${var})
-        get_filename_component(SHORT_NAME "${LIST_DIR}" NAME)
-        message("Including custom CMakeLists.txt ${SHORT_NAME}")
-        add_subdirectory(${LIST_DIR} ${SHORT_NAME})
-    endforeach()
-endmacro()
+    macro(pico_sdk_init)
+        message("SDK INIT ${PICO_SDK_PATH}")
```
Comment (lurch) [inline pico_sdk_init.cmake:43]: I guess the `PICO_SDK_PATH` is already printed by line 34, so should this be a DEBUG message? :shrug:

Comment (kilograham) [reply pico_sdk_init.cmake:43]: ah that was some left over debugging, thanks

**Review by liamfraser — APPROVED** [review]

File: src/common/pico_sync/sem.c [hunk 4]
```diff
@@ -6,6 +6,7 @@
 
 #include "pico/sem.h"
 #include "pico/time.h"
+#include "sys/select.h"
```
Comment (lurch) [inline src/common/pico_sync/sem.c:9]: ???
This PR seems like a strange place for this change to be included? :wink:

Comment (kilograham) [reply src/common/pico_sync/sem.c:9]: doh! i was looking at something else, and was wondering if it was available

**Pull request merged**